### ISSUE TITLE
[BUG]: The `get-architectures` job does not support projects that don't specify the `architectures` and `platforms` property

### ISSUE THREAD

**Issue by brlin-tw** [snapcrafters/ci#44]
### What changes would you like?

Currently, the `get-architectures` workflow will return an empty array on projects that have snapcraft.yaml not declaring any `platforms` and `architectures`, making the `release-to-candidate` workflow error with the following message:

```txt
Error when evaluating 'strategy' for job 'release'. .github/workflows/release-to-edge.yaml (Line: 36, Col: 23): Matrix vector 'architecture' does not contain any values
```

It should rather return the full list of architectures that Launchpad supports building on that base, which matches the behavior of snapcraft.io/build

**Comment by soumyaDghosh** [comment]
We use `remote-build` for building. Enabling such builds which will never successfully is kind of unnecessary and uses unnecessary resources IMO.

**Comment by jnsgruk** [comment]
Agreed - this is by design. We require explicit declaration of the desired target architectures.
